### ISSUE TITLE
[CGNS-271] OSx: Metal version of cgnsplot, i.e. move from OpenGL to Metal

### ISSUE THREAD

**Issue by brtnfld** [CGNS/CGNS#666]
> This issue has been imported from JIRA. Read the [original ticket here](https://cgnsorg.atlassian.net/browse/CGNS-271).

- _**Created at:**_ Wed, 9 Mar 2022 09:46:18 -0600

<p>Since Apple has deprecated OpenGL in favor of Metal, a Metal version of cgnstools using OpenGL should be developed for Metal. This should result in a substantial performance improvement.</p>

**Comment by brtnfld** [comment]
> This comment has been imported from JIRA. Read the [original comment here](https://cgnsorg.atlassian.net/browse/CGNS-271#comment-14143).

- _**Created at:**_ Fri, 15 Jul 2022 12:14:42 -0500

<p>If we really plan on moving away from OpenGL, moving away from tcl should also be first considered. A similar move from OpenGL to Vulkan could be nice on unix systems.</p>

**Comment by brtnfld** [comment]
A newer alternative would be to use https://chromium.googlesource.com/angle/angle, which would avoid targeting a specific back end.

**Comment by brtnfld** [comment]
@MicK7, I'm looking into this and dawn does have it advantages, mainly corporate backing and open standard. But bgfx, https://github.com/bkaradzic/bgfx, seems to be more lightweight and straightforward, and probably a better fit. However, since a single lead developer primarily drives it, it is dependent on that developer's commitment. Thoughts?

**Comment by MicK7** [comment]
I agree dawn is clearly not straighforward. Recently, I was looking into a small similar project and the lib selection was instead:  Nuklear (for widgets) and SDL3_gpu (for rendering abstraction) . Both are quite mainstream and have C API.
Thanks for the discovery of bgfx.
It seems like bgfx may have great abstraction layers and be easier to play with in C++. From my quick search , people who use bgfx tends to stick with a tagged version because it is good enough for their usecase. So the single lead developper risk can't be denied but may not be so impactful. I have not found recent big projects using bgfx though. Otherwise, as developpement tools are quickly evolving with IA, I am not sure that moving to another library in 5 to 10 years from now will be an issue.
My two cents ^^

**Comment by brtnfld** [comment]
Thanks for the feedback, I'll look into bgfx as a start.

**Comment by brtnfld** [comment]
@MicK7, I've been looking into bgfx, and so far, I've not been able to get it to work. The issue is that this (and likely most of them like it) is not really meant for TK/TCL. I've not been able to find a current TK/TCL application that is using them. I'll spend a little more time investigating, but I've only so far been able to get a tangled mess of lines within a window inside a window.
